presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -139,7 +146,44 @@ public void createTable(Table table)
     public void dropTable(String databaseName, String name, boolean deleteData)
             throws TException
     {
+        Table table = client.get_table(databaseName, name);
+
         client.drop_table(databaseName, name, deleteData);
+
+        String tableLocation = table.getSd().getLocation();
+        if (deleteData && isManagedTable(table) && !isNullOrEmpty(tableLocation)) {
+            // Workaround for issue https://github.com/prestosql/presto/issues/1053
+            deleteTableFilesIfNeeded(databaseName, name, new Path(tableLocation));
+        }
+    }
+
+    private void deleteTableFilesIfNeeded(String databaseName, String tableName, Path tablePath)
+    {
+        try {
+            if (isS3Table(tablePath)) {
+                PrestoS3FileSystem fs = new PrestoS3FileSystem();
```

Comment:
@rohangarg I guess I need to inject it somehow

Reply:
First I will move the logic to `io.prestosql.plugin.hive.metastore.thrift.ThriftHiveMetastore` and use `thriftConfig`

Reply:
Yes, that can be done. Although, I am not sure about why it can't be done in `SemiTransactionalHiveMetastore#prepareDropTable`.

Reply:
> Yes, that can be done. Although, I am not sure about why it can't be done in `SemiTransactionalHiveMetastore#prepareDropTable`.

It might be possible, but I found it complex  since it's running un-synchronously.

Reply:
Looked up a bit more, and the delete shouldn't be done by presto because in hive for managed tables, it should be handled by metastore only.

Reply:
> Looked up a bit more, and the delete shouldn't be done by presto because in hive for managed tables, it should be handled by metastore only.

That's true, but since files are not deleted, I suggest this solution (same as implemented in Glue metastore)